iconset-builder.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+var path = require('path'),
+	q = require('q'),
+	xml2js = require('xml2js'),
+	fs = require('fs');
+
+var iconsetObjTemplate = {
+	'link': [
+		{'$': {'rel': 'import', 'href': '../polymer/polymer.html'}},
+		{'$': {'rel': 'import', 'href': '../iron-iconset-svg/iron-iconset-svg.html'}},
+	],
+	'iron-iconset-svg': {
+		'$': {'name': '', 'size': ''},
+		'svg': {'defs': {'g': []}}
+	}
+};
+
+function processFill(elem) {
+	if (elem.$ && elem.$.fill && elem.$.fill !== 'none') {
+		delete elem.$.fill;
+	}
+	for(var child in elem) {
+		if (child !== '$') {
+			processFill(elem[child]);
+		}
+	}
+}
+
+function buildFileXml(iconsetObj, iconsetPath, file) {
+
+	var deferred = q.defer();
+
+	var extension = path.extname(file);
+	if (extension !== '.svg') {
+		return;
+	}
+
+	fs.readFile(path.join(iconsetPath, file), 'utf8', function(err, xml) {
+		if (err) {
+			return deferred.reject(err);
+		}
+		var parser = new xml2js.Parser();
+		parser.parseString(xml, function(err, result) {
+			if (err) {
+				return deferred.reject(err);
+			}
+			if (iconsetObj['iron-iconset-svg'].$.size === '' && result.svg.$.viewBox) {
+				var values = result.svg.$.viewBox.split(' ');
+				iconsetObj['iron-iconset-svg'].$.size = values[values.length-1].toString();
```

Comment:
If I understand this correctly, we assume the image dimensions are square?  I suspect that's probably safe?  Also, I am guessing all the images we get from graphics folks can't have viewBox offsets, or we need to ensure those are corrected before putting into repo?